core/src/main/scala/schema/provider/metadata/SchemaConversions.scala

```diff
@@ -0,0 +1,200 @@
+    } leftMap (SchemaDefinitionProviderError(_))
+
+  def toCreateTable(metadata: AbstractTableMetadata): SchemaResult[CreateTable] =
+    Either.catchNonFatal {
+      for {
+        columns <- metadata.getColumns.asScala.toList.traverse(toTableColumn)
+        primaryKey <- toPrimaryKey(
+          metadata.getPartitionKey.asScala.toList,
+          metadata.getClusteringColumns.asScala.toList)
+      } yield
+        CreateTable(
+          ifNotExists = false,
+          tableName = TableName(Some(KeyspaceName(metadata.getKeyspace.getName)), metadata.getName),
+          columns = columns,
+          primaryKey = Some(primaryKey),
+          options = Seq.empty
+        )
+    } leftMap (SchemaDefinitionProviderError(_)) joinRight
+
+  def readTable(metadata: IndexMetadata): TableName =
+    TableName(Some(KeyspaceName(metadata.getTable.getKeyspace.getName)), metadata.getTable.getName)
+
+  def toCreateIndex(
+      metadata: IndexMetadata,
+      readTable: (IndexMetadata) => TableName = readTable): Either[
+    SchemaDefinitionProviderError,
+    CreateIndex] =
+    Either.catchNonFatal {
+      CreateIndex(
+        isCustom = metadata.isCustomIndex,
+        ifNotExists = false,
+        indexName = Option(metadata.getName),
+        tableName = readTable(metadata),
+        identifier = Index.Identifier(metadata.getTarget),
+        using =
+          if (metadata.isCustomIndex)
+            // The options are not visible in the IndexMetadata class
+            Some(Index.Using(metadata.getIndexClassName, None))
+          else None
+      )
+    } leftMap (SchemaDefinitionProviderError(_))
+
+  def toUserType(userType: UserType): SchemaResult[CreateType] =
+    Either.catchNonFatal {
+      userType.getFieldNames.asScala.toList.traverse { fieldName =>
+        toField(fieldName, userType.getFieldType(fieldName))
+      } map { list =>
+        CreateType(
+          ifNotExists = false,
+          typeName = TypeName(Some(KeyspaceName(userType.getKeyspace)), userType.getTypeName),
+          fields = list)
+      }
+    } leftMap (SchemaDefinitionProviderError(_)) joinRight
+
+  private[this] def toField(
+      name: String,
+      datastaxDataType: DatastaxDataType): SchemaResult[Field] =
+    toDataType(datastaxDataType) map { dataType =>
+      Field(name, dataType)
+    }
+
+  private[this] def toTableColumn(metadata: ColumnMetadata): SchemaResult[Table.Column] =
+    toDataType(metadata.getType).map { dataType =>
+      Table.Column(
+        name = metadata.getName,
+        dataType = dataType,
+        isStatic = metadata.isStatic,
+        isPrimaryKey = false)
+    }
+
+  private[this] def toDataType(dataType: DatastaxDataType): SchemaResult[DataType] = {
+
+    import DatastaxDataType._
+
+    def toDataTypeNative(dataType: DatastaxDataType): SchemaResult[DataType.Native] =
+      dataType.getName match {
+        case Name.ASCII     => DataType.Ascii.asRight
```

Comment:
Where is `asRight` defined? `as` suually denotes an internal cast and if this is is just lifting the value into `Either.Right` we should use `toRight` or the `right()` syntax from cats.

Reply:
`asRight` is the new syntax in Cats (see PR https://github.com/typelevel/cats/pull/1454).

`asRight` and `asLeft` where chosen so they can't be confused with the `.right` and `.left` methods on `Either` for `RightProjection` and `LeftProjection`.